### PR TITLE
DAOS-6346 test: Prepare dcp to be backwards compatible

### PR DESCRIPTION
**Pull request by daltonbohning**
Adjust the DataMover command to be compatible with the current
version of dcp, and the upcoming new version, by querying dcp
to see which options are available.
If dcp does not have the svcl options, they are removed from the
command.
If dcp does not have the daos-src-pool, etc., the format is updated
to instead use daos://pool/cont/path

Skip copy_bad_params until the interface changes are stable.
Will be re-enabled by DAOS-6355

Signed-off-by: Dalton Bohning <daltonx.bohning@intel.com>

### CODE REVIEW

**Comment by daltonbohning** [comment]
Step 1: Run compatibility changes against current version of mpifileutils in CI.
All datamover tests completed successfully:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-4126/3/tests

**Comment by daltonbohning** [comment]
Step 1: Run compatibility changes against current version of mpifileutils in CI.
**_DONE_**
All datamover tests completed successfully:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-4126/3/tests

Step 2: Run compatibility changes against this updated version of mpifileutils in CI.
**_DONE_**
All datamover tests completed successfully:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-4126/6/tests

**Comment by daltonbohning** [comment]
Step 1: Run compatibility changes against current version of mpifileutils in CI.
**_DONE_**
All datamover tests completed successfully:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-4126/3/tests

Step 2: Run compatibility changes against the updated version of mpifileutils in CI.
**_DONE_**
All datamover tests completed successfully:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-4126/6/tests

Latest run with no pragmas is successful:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-4126/8/pipeline/748/

File: src/tests/ftest/util/data_mover_utils.py [hunk 1]
```diff
@@ -131,14 +131,32 @@ def set_datamover_params(self, src_pool=None, dst_pool=None, src_cont=None,
...
-    def __init__(self, hosts, timeout=30):
+    def __init__(self, hosts, tmp, timeout=30):
         """Create a datamover object."""
         super(DataMover, self).__init__(
             "/run/datamover/*", "dcp")
 
         # set params
         self.timeout = timeout
         self.hosts = hosts
+        self.tmp = tmp
```
Comment (shimizukko) [inline src/tests/ftest/util/data_mover_utils.py:142]: Maybe this is out of scope, but `tmp` doesn't look like a good variable name.

**Review by shimizukko — APPROVED** [review]
LGTM

**Review by brianjmurrell — COMMENTED** [review]
Seems reasonable to me but I will reserve my +1 for somebody more familiar with this test code.

Great work @daltonbohning on implementing a change to maintain backword compatibility without introducing a _flag day_ change that would cause others' in-flight PRs to break.

File: src/tests/ftest/util/data_mover_utils.py (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/util/data_mover_utils.py:142]: I don't disagree, but `self.tmp` comes from deeper in the test framework, such as:
https://github.com/daos-stack/daos/blob/89a9fa8f0172a6ded9ba2dbdd41d955ae09d6a27/src/tests/ftest/util/apricot/apricot/test.py#L220
And then various utils use `self.tmp`.

**Comment by daltonbohning** [comment]
> Seems reasonable to me but I will preserve the +1 for somebody more familiar with this test code.
> 
> Great work @daltonbohning on implementing a change to maintain backword compatibility without introducing a _flag day_ change that would cause others' in-flight PRs to break.

Sure, and thanks! I mostly wanted you to be aware of this PR because you will probably be the one to approve/reject the mpifileutils-pkg PR :)

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Pull request merged**